terraform/aws/modules/rds-aurora-postgresql/main.tf

```diff
@@ -46,26 +46,28 @@ data "aws_vpc" "provisioning_vpc" {
-  engine_version                  = var.engine_version
-  kms_key_id                      = aws_kms_key.aurora_storage_key.arn
-  master_username                 = var.username
-  master_password                 = local.master_password
-  final_snapshot_identifier       = "${var.final_snapshot_identifier_prefix}-${format("rds-cluster-multitenant-%s-%s", split("-", var.vpc_id)[1], local.database_id)}"
-  skip_final_snapshot             = var.skip_final_snapshot
-  deletion_protection             = var.deletion_protection
-  backup_retention_period         = var.backup_retention_period
-  preferred_backup_window         = var.preferred_backup_window
-  preferred_maintenance_window    = var.preferred_maintenance_window
-  port                            = var.port
-  db_subnet_group_name            = "mattermost-provisioner-db-${var.vpc_id}-postgresql"
-  vpc_security_group_ids          = [data.aws_security_group.db_sg.id]
-  storage_encrypted               = var.storage_encrypted
-  apply_immediately               = var.apply_immediately
-  db_cluster_parameter_group_name = aws_rds_cluster_parameter_group.cluster_parameter_group_postgresql.id
-  copy_tags_to_snapshot           = var.copy_tags_to_snapshot
-  snapshot_identifier             = var.creation_snapshot_arn == "" ? null : var.creation_snapshot_arn
+  cluster_identifier               = format("rds-cluster-multitenant-%s-%s", split("-", var.vpc_id)[1], local.database_id)
+  engine                           = var.engine
+  engine_version                   = var.engine_version
+  kms_key_id                       = aws_kms_key.aurora_storage_key.arn
+  master_username                  = var.username
+  master_password                  = local.master_password
+  final_snapshot_identifier        = "${var.final_snapshot_identifier_prefix}-${format("rds-cluster-multitenant-%s-%s", split("-", var.vpc_id)[1], local.database_id)}"
+  skip_final_snapshot              = var.skip_final_snapshot
+  deletion_protection              = var.deletion_protection
+  backup_retention_period          = var.backup_retention_period
+  preferred_backup_window          = var.preferred_backup_window
+  preferred_maintenance_window     = var.preferred_maintenance_window
+  port                             = var.port
+  db_subnet_group_name             = "mattermost-provisioner-db-${var.vpc_id}-postgresql"
+  vpc_security_group_ids           = [data.aws_security_group.db_sg.id]
+  storage_encrypted                = var.storage_encrypted
+  apply_immediately                = var.apply_immediately
+  db_cluster_parameter_group_name  = aws_rds_cluster_parameter_group.cluster_parameter_group_postgresql.id
+  db_instance_parameter_group_name = aws_db_parameter_group.db_parameter_group_postgresql.id
+  copy_tags_to_snapshot            = var.copy_tags_to_snapshot
+  snapshot_identifier              = var.creation_snapshot_arn == "" ? null : var.creation_snapshot_arn
+  allow_major_version_upgrade      = true
```

Comment:
@andrleite make it variable as we want mostly to be false to avoid an accidental update

Reply:
@stafot Do you think we need to expose this parameter through dbfactory API?